test/model_api.jl

```diff
@@ -95,3 +94,47 @@ mutable struct UnivariateFiniteFitter <: Probabilistic end
     @test yhat == fill(DummyUnivariateFinite(), 3)
 
 end
+
+@testset "fallback for `report()` method" begin
+    report_given_method =
+        OrderedCollections.OrderedDict(
+            :predict=>(y=7,),
+            :fit=>(x=1, z=3),
+            :transform=>nothing,
+        )
+    @test MLJModelInterface.report(APIx0(f0=1), report_given_method) ==
+        (x=1, z=3, y=7)
+
+    report_given_method =
+        OrderedCollections.OrderedDict(
+            :predict=>(y=7,),
+            :fit=>(y=1, z=3),
+            :transform=>nothing,
+        )
+    @test MLJModelInterface.report(APIx0(f0=1), report_given_method) ==
+        (y=1, z=3, predict=(y=7,))
+
+    @test MLJModelInterface.report(
+        APIx0(f0=1),
+        OrderedCollections.OrderedDict(:fit => nothing, :transform => NamedTuple()),
+    ) |> isnothing
+
+    @test MLJModelInterface.report(
+        APIx0(f0=1),
+        OrderedCollections.OrderedDict(:fit => 42),
+    ) == 42
+
+    @test MLJModelInterface.report(
+        APIx0(f0=1),
+        OrderedCollections.OrderedDict(:fit => nothing),
+    ) |> isnothing
+
+    @test MLJModelInterface.report(
+        APIx0(f0=1),
+        OrderedCollections.OrderedDict(:fit => NamedTuple()),
```

Comment:
I guess this returns `report = nothing` which would be breaking for the end user right?

Reply:
Yeah, you're right. 

Although this does not in itself break MLJBase (or anything else) the companion MLJBase PR I have made, which makes use of this new `report` method, will break some existing uses: Previously `report(mach)` returned `NamedTuple()` and now it will return `nothing`. And I did have to tweak some MLJBase tests to account for this change in that companion PR. So will need to flag that PR as (mildly) breaking. 

I am already resolved to the fact that a follow-up to the MLJBase PR (in which canned composites like Stack and Pipeline are re-implemented using the new `NetworkComposite` type) will need to be tagged as breaking since composite reports are going to look a bit different. This can all be combined into one release.

Reply:
That's fine. I just wanted to be sure you were aware of the change.